tartufo/scanner.py

```diff
@@ -463,20 +463,28 @@ def chunks(self) -> Generator[types.Chunk, None, None]:
 
         try:
             if self.git_options.branch:
-                branches = self._repo.remotes.origin.fetch(self.git_options.branch)
+                # Single branch only
+                unfiltered_branches = list(self._repo.branches)
+                branches = [
+                    x for x in unfiltered_branches if x == self.git_options.branch
+                ]
+                if self.git_options.fetch:
+                    self._repo.remotes.origin.fetch(self.git_options.branch)
             else:
-                branches = self._repo.remotes.origin.fetch()
+                # Everything
+                branches = list(self._repo.branches)
+                if self.git_options.fetch:
+                    self._repo.remotes.origin.fetch()
```

Comment:
We should probably invert these blocks and first fetch, then list `_repo.branches` in case that list is now longer, right?

Reply:
We should probably also iterate over the remotes instead of hard-coding origin, or have a list of remotes to fetch as a parameter, but that could be a different PR?

Reply:
I agree, I'll reverse the order of fetch and branch listing.

As far as not hard coding origin, I strongly agree but I'll make that a different PR since that's the existing behavior prior to this PR.

Reply:
Yeah, the assumption of `origin` is a bug that has yet to be filed...